### PR TITLE
A14.1 - Anagrams

### PR DESCRIPTION
**Pull request by vikrampg1710-tmm**
**A14.1:** Printing all possible anagrams from dictionary in O(n) time.

### CODE REVIEW

**Review by prasannameta — CHANGES_REQUESTED** [review]
Please add comments.

File: Assignments/Program.cs [hunk 1]
```diff
@@ -11,34 +11,39 @@
...
 using System.IO;
 using System.Diagnostics;
+using System.Reflection;
 
 namespace Academy;
 public class A14 {
    public static void Main () {
-      var wordList = new List<string> (File.ReadAllLines (@"c:/etc/words.txt"));
+      // Loading all the dictionary words from a assembly-manifest resource file.
+      using var reader = new StreamReader (Assembly.GetExecutingAssembly ().GetManifestResourceStream ($"Assignments.Files.words.txt"));
```
Comment (prasannameta) [inline Assignments/Program.cs]: You could have used File.ReadAllLines like taught in class. Why do you want to embed this file with project assembly.

File: Assignments/Program.cs [hunk 2]
```diff
@@ -11,34 +11,39 @@
...
-         foreach (char c in word) array[c - 65]++;
-         return string.Join ("", array);
-      }
    }
 
+   /// <summary>Returns an unique string made of hash-table digits of given input string</summary>
+   // Basically, this method returns a string whose length is 26 with chars either '1' or '0'.
+   // The char at Nth index in the returned string represents the presence/absence of (N+1)th alphabet in the given word given word.
+   // So the char '1' at 0th index in the returned string implies that 1st alphabet 'A' is present in the given string & vice versa.
+   static string HashCode (ref int[] array, string word) {
```
Comment (prasannameta) [inline Assignments/Program.cs]: Please explain why use ref in your most function parameters.

**Review by prasannameta — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 1 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: I thought making the file as an embedded resource will make the code faster.

File: Assignments/Program.cs (hunk 2 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: This way we can reuse the same array created in the Main method, many times.

Comment (prasannameta) [reply Assignments/Program.cs]: Please see the answer from Jon Skeet: https://stackoverflow.com/questions/635915/when-to-use-ref-and-when-it-is-not-necessary-in-c-sharp

**Review by prasannameta — APPROVED** [review]

**Pull request closed without merging**